### DISCUSSION TITLE
How to disable out-of-spec updates?

### DISCUSSION THREAD

**Discussion by mrkamel** [renovatebot/renovate#13347] [Request Help]
We have a ruby on rails repo and a dependency spec for rails in our Gemfile `~> 6.0`, meaning `>= 6.0` and `< 7`. Moreover, we use `rangeStrategy: update-lockfile` and `updatePinnedDependencies: false`. The issue we face is that renovate tries to update rails to `7.0` but we want out-of-spec updates to be disabled completely. Can this be done somehow? The other values for `rangeStrategy` don't seem to fit either. Thanks in advance.

**Comment by HonkingGoose** [comment]
I suspect something like this could work.

`matchPackageNames` specifies that the rule is only valid for the `rails` package.
`matchManagers` is used to limit the rule further to only packages that get updated via the `bundler` package manager.

```json
{
  "packageRules": [
    {
      "matchPackageNames": ["rails"],
      "matchManagers": ["bundler"],
      "allowedVersions": "<=7"
    }
  ]
}
```

Docs: 

- https://docs.renovatebot.com/configuration-options/#matchpackagenames
- https://docs.renovatebot.com/configuration-options/#matchmanagers
- https://docs.renovatebot.com/configuration-options/#allowedversions

**Reply by mrkamel** [reply]
yes, thx, this is package specific. i'm rather searching for a general solution to disable out-of-spec updates

**Reply by rarkins** [reply]
There is no solution for this currently, but we've thought about it in #13050. For now, just close the v7 PR and it will be ignored forever though (until there's a v8).

In general I'd consider *disabling* such updates a bad idea because it increases the chances that you are behind in dependencies and unaware of it. It would be better to set `dependencyDashboardApproval` to `true` instead of disabling so that you at least have visibility of what you're missing.
